### DISCUSSION TITLE
Terminate Webhook SSL on Load balancer

### DISCUSSION THREAD

**Discussion by g1soori** [actions/actions-runner-controller#1889] [Questions]
As per our setup, we are terminating SSL connections at AWS ALBs. Is there way to run webhook server in http instead of https ?

**Comment by chrisforrette** [comment]
@g1soori I got this working recently and I'm happy to share what we did...

We're running self-hosted runners with this controller using an ALB and EKS and for the load balancer we have:
* An internet-facing ALB (we put Cloudflare DNS in front of this along with a WAF rule restricting access to GitHub webhook CIDR ranges)
* A 443 listener on that ALB wired up with an SSL certificate from AWS ACM
* A rule on that listener for path `/webhook` pointing to a target group
* The target group uses HTTP protocol for both traffic and the health check, default health check path of `/`, and an IP target type pointing at the webhook service pods

Let me know if there's any other detail that would be helpful!
